pkg/app/build.go

```diff
@@ -123,6 +124,23 @@ func build(clicontext *cli.Context) error {
 	return BuildImage(clicontext, builder)
 }
 
+func DetectEnvironment(clicontext *cli.Context, buildOpt builder.Options) error {
+	engine, err := envd.New(clicontext.Context, "docker")
+	if err != nil {
+		return errors.Wrap(err, "failed to create the docker client")
+	}
+	// detect if the current environment is running before building
+	ctr := filepath.Base(buildOpt.BuildContextDir)
+	running, _ := engine.IsRunning(clicontext.Context, ctr)
+	if err != nil {
+		return err
+	}
+	if running {
+		logrus.Warnf("detect container %s is running.", ctr)
```

Comment:
I think it should return an error if `--force` is not set. Because it will fail later.

Reply:
> I think it should return an error if `--force` is not set. Because it will fail later.
If so I think this part of the logic overlaps with the logic of func (e dockerEngine) CleanEnvdIfExists(ctx context.Context, name string, force bool) error I tend to remove the code in CleanEnvdIfExists.

Reply:
The modified code looks like this.
```
func (e dockerEngine) CleanEnvdIfExists(ctx context.Context, name string, force bool) error {
	created, err := e.Exists(ctx, name)
	if err != nil {
		return err
	}
	if !created {
		return nil
	}
	return e.ContainerRemove(ctx, name, dockertypes.ContainerRemoveOptions{Force: force})
}
```

Reply:
cc @gaocegege should we bring the `CleanEnvdIfExists` before `BuildImage`?

`CleanEnvdIfExists` and `DetectEnvironment` are doing something similar.

Reply:
Personally, prefer removing the code in CleanEnvdIfExists.

The func `CleanEnvdIfExists` is defined in envd.Engine. And the engine interface will have two implementations soon. Thus it will be great to keep the logic there simple and clean.

WDYT @kemingy

Reply:
@kemingy  @gaocegege I tried to accomplish this modification.